src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,56 @@
+@import "../variables/variables";
+
+.calendar {
+  margin: 100px auto;
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  width: 710px;
+  align-items: center;
+
+  &__day {
+    @include field-size($fields-size);
+    border: 1px solid black;
+    box-sizing: border-box;
+    background-color: #eee;
+    font-family: $font-family;
+    font-size: $font-size;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    transition: all ease-in-out 0.5s;
+
+    &:hover {
+      cursor: pointer;
+      background-color: pink;
+      transform: translateY(-20px);
+    }
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i}) {
+        &::before {
+          content: "#{$i}";
+          font-family: $font-family;
+          font-size: $font-size;
+        }
+      }
+    }
+  }
+}
+@each $day, $glyph in $start-day {
+  .calendar--start-day-#{$day} {
+    .calendar__day {
+      &:first-child {
+        margin-left: #{$glyph}px;
```

Comment:
Calculate the margin basing on your variables. I left a couple more comments about it before.